touchforms/formplayer/static/formplayer/script/entrycontrols_full.js

```diff
@@ -1,785 +1,584 @@
-      var widget = $container.find('#textfield');
-    } else {
-      $container.html('<textarea id="textarea" style="width: 33em; height: 10em; font-family: sans-serif;"></textarea>');
-      var widget = $container.find('#textarea');
-    }
-    this.inputfield = widget[0];
-    widget.change(function() { q.onchange(); });
-  }
-
-  this.getControl = function () {
-    return this.inputfield;
-  }
-
-  this.getRaw = function () {
-    var control = this.getControl();
-    return (control != null ? control.value : null);
-  }
-
-  this.getAnswer = function () {
-    var raw = $.trim(this.getRaw());
-    return (raw == '' ? null : raw);
-  }
-
-  this.setAnswer = function (answer, postLoad) {
-    var control = this.getControl();
-    if (control) {
-      control.value = (answer != null ? answer : '');
-    } else {
-      this.default_answer = answer;
-    }
-  }
-
-  this.prevalidate = function (q) {
-    var raw = this.getRaw();
-    if (raw) {
-      var errmsg = this._prevalidate(raw);
-      if (errmsg) {
-        q.showError(errmsg);
-        return false;
-      }
-    }
-    return true;
-  }
 
-  this._prevalidate = function (raw) {
-    return null;
-  }
+/**
+ * The entry that defines an float input. Only accepts real numbers
+ */
+function FloatEntry(question, options) {
+    FreeTextEntry.call(this, question, options);
+    this.templateType = 'float';
 
-  this.domainText = function() {
-    return 'free-text';
-  }
+    this._prevalidate = function(raw) {
+        return (isNaN(+raw) ? "Not a valid number" : null);
+    }
 
-  this.widgetWidth = function() {
-    return '20em';
-  }
+    this.domainText = function() {
+        return 'decimal';
+    }
 }
+FloatEntry.prototype = Object.create(IntEntry.prototype);
+FloatEntry.prototype.constructor = IntEntry;
+
+/**
+ * Represents a checked box entry.
+ */
+function MultiSelectEntry(question, options) {
+    var self = this;
+    Entry.call(this, question, options);
+    self.templateType = 'select';
+    self.choices = question.choices;
+    self.choicevals = question.choicevals ? question.choicevals() : null;
+    self.answer = question.answer;
 
-function PasswordEntry (args) {
-  args.length_limit = args.length_limit || 9;
-  inherit(this, new FreeTextEntry(args));
+    self.group = nonce();
 
-  this.mkWidget = function () {
-    $('#answer')[0].innerHTML = '<input id="textfield" maxlength="' + this.length_limit + '" type="passwd"/>';
-    this.inputfield = $('#textfield')[0];
-  }
-}
+    self.isMulti = true;
+    self.default_selections = null;
 
-function IntEntry (parent, length_limit) {
-  inherit(this, new FreeTextEntry({parent: parent, domain: 'numeric', length_limit: length_limit || 9}));
+    self.onClear = function() {
+        self.answer([]);
+    };
 
-  this.getAnswer = function () {
-    var val = this.parent('getAnswer')();
-    return (val != null ? +val : val);
-  }
+    self.COMMIT_DELAY = 300; //ms
+    self.pendchange = function(immed) {
```

Comment:
this should be refactored and brought to the adapter level for a more generic throttling of all `answer-question`